src/Elm.elm

```diff
@@ -1156,6 +1156,135 @@ lambda argBaseName argType toExpression =
         }
 
 
+{-| -}
+lambdaBetaReduced : String -> Elm.Annotation.Annotation -> (Expression -> Expression) -> Expression
+lambdaBetaReduced argBaseName argType toExpression =
+    let
+        arg1 =
+            valueWith [] argBaseName argType
+
+        (Compiler.Expression expr) =
+            toExpression arg1
+    in
+    Compiler.Expression
+        { expression =
+            betaReduce <|
```

Comment:
This is the only line that differs from `lambda`, so there is some redundant code in those two functions